crates/compiler/gen_dev/src/generic64/aarch64.rs

```diff
@@ -477,23 +473,26 @@ impl CallConv<AArch64GeneralReg, AArch64FloatReg, AArch64Assembler> for AArch64C
         aligned_stack_size: i32,
         fn_call_stack_size: i32,
     ) {
+        let frame_pointer_link_register = 16;
+
         if aligned_stack_size > 0 {
             // All the following stores could be optimized by using `STP` to store pairs.
-            let mut offset = aligned_stack_size;
-            offset -= 8;
-            AArch64Assembler::mov_reg64_stack32(buf, AArch64GeneralReg::LR, offset);
-            offset -= 8;
-            AArch64Assembler::mov_reg64_stack32(buf, AArch64GeneralReg::FP, offset);
+            let mut offset = aligned_stack_size - fn_call_stack_size - frame_pointer_link_register;
 
-            offset = aligned_stack_size - fn_call_stack_size;
             for reg in saved_general_regs {
+                AArch64Assembler::mov_reg64_base32(buf, *reg, -offset);
                 offset -= 8;
-                AArch64Assembler::mov_reg64_base32(buf, *reg, offset);
             }
+
             for reg in saved_float_regs {
+                AArch64Assembler::mov_freg64_base32(buf, *reg, -offset);
                 offset -= 8;
-                AArch64Assembler::mov_freg64_base32(buf, *reg, offset);
             }
+
+            let w = aligned_stack_size;
+            AArch64Assembler::mov_reg64_stack32(buf, AArch64GeneralReg::FP, w - 0x10);
+            AArch64Assembler::mov_reg64_stack32(buf, AArch64GeneralReg::LR, w - 0x08);
+
```

Comment:
@lukewilliamboswell there was a bug here where we mov above relative to the base pointer, but the base pointer was already updated to its earlier (it's caller's) value. x86 does things differently, but still I think we should review this code together sometime to convince ourselves that everything is correct